### ISSUE TITLE
pattern.indexOf is not a function

### ISSUE THREAD

**Issue by kkprince1998** [gruntjs/grunt-contrib-jade#150]
I gotz an error:
```
Warning: pattern.indexOf is not a function Use --force to continue.
```

#### MORE INFO:
Here is my jade task:

        jade: {
            options: {
                pretty: true
            },
            all: {
                files: {
                    expand:true,
                    cwd: 'src/static/jade',
                    ext: "html",
                    src: ['src/static/jade/**/*.jade', '!src/static/jade/_includes'],
                    dest: 'build/'
                }
            }
        }

So basically I am trying to take the jade files in `src/static/jade` (including subdirs, except `_include`) and put them in `build`, keeping the directory structure. I have tryed commenting the `expand` line, however it gives me:

     Warning: Unable to read "src/static/jade" file (Error code: EISDIR). Use --force to continue.

Perhaps I am going about this the wrong way. How should I fix this?

I also posted this on stackoverflow: http://stackoverflow.com/questions/33092564/grunt-jade-error

**Comment by kkprince1998** [comment]
My files array was an obj.

**Comment by alubeck** [comment]
Yeahh, your comment let me check my Gruntfile again. I had the same issue, thanks for inspiration!

**Comment by arpit1997** [comment]
That was a very good comment @KaidenP 
:small_airplane: :small_blue_diamond:

**Comment by KaidenP** [comment]
Thank You?

**Comment by kb2412** [comment]
you could try write that:

jade: {
        options: {
            pretty: true
        },
        all: {
            expand:true,
            cwd: 'src/static/jade',
            ext: "html",
            src: ['src/static/jade/**/*.jade', '!src/static/jade/_includes'],
            dest: 'build/'
        }
    }

**Comment by C0ZEN** [comment]
The error is pretty obvious after the solution was found...
`.length` is for array, so the `files` must be an array, not an object.
